packages/cli/src/create-app/runNpmInstall.cjs

```diff
@@ -0,0 +1,83 @@
+const { spawnSync } = require( 'child_process' );
+const ora = require( 'ora' ); // Add ora for spinner
+
+/**
+ * Runs npm install in the project directory with improved output handling.
+ *
+ * @param {string} projectDirPath - Path to the project directory.
+ * @param {Object} options - Configuration options.
+ * @param {boolean} options.skipInstall - Whether to skip the installation step.
+ * @return {Promise<boolean>} - Returns true if installation was successful or skipped.
+ */
+const runNpmInstall = async ( projectDirPath, options = {} ) => {
```

Comment:
Bad Pattern. If `--skip-install` just don't call this function in the main process.